easybuild/easyblocks/t/tensorflow.py

```diff
@@ -152,6 +153,135 @@ def write_wrapper(self, wrapper_dir, compiler, i_mpi_root):
+
+        The result is a list of tuples (tf_name, eb_name) where eb_name can be None if no EC exists or True to assume
+        the dependency is always installed (e.g. via pip in the TF EC)
+
+        Note: Can be used to check against third_party/systemlibs/syslibs_configure.bzl by running:
+              from easybuild.easyblocks.tensorflow import EB_TensorFlow
+              print(EB_TensorFlow.get_system_libs_for_version('2.1.0', as_VALID_LIBS=True))
+        """
+        tf_version = LooseVersion(tf_version)
+
+        def is_version_ok(version_requirement):
+            """Return True if the TF version to be installed matches the version_requirement"""
+            min_version = LooseVersion(version_requirement[0])
+            result = False
+            if tf_version >= min_version:
+                # min. version matches, so OK if max version matches or not present
+                if len(version_requirement) == 1 or tf_version < LooseVersion(version_requirement[1]):
+                    result = True
+            return result
+
+        # For this list check third_party/systemlibs/syslibs_configure.bzl --> VALID_LIBS
+        # Also verify third_party/systemlibs/<name>.BUILD if it does something "strange" (e.g. link hardcoded headers)
+        available_system_libs = (
+            # Format: 1. name in TF
+            #         2. dependency name in EB, can use None if unknown or True to assume installed
+            #         3. min. (incl), max. (excl) TF version required
+            ('absl_py', True, ('2.1.0',)),
+            ('astor_archive', True, ('2.1.0',)),
+            ('astunparse_archive', True, ('2.2.0',)),
+            ('boringssl', None, ('2.1.0',)),
+            ('com_github_googleapis_googleapis', None, ('2.1.0',)),
+            ('com_github_googlecloudplatform_google_cloud_cpp', None, ('2.1.0',)),  # Not used due to $TF_NEED_GCP=0
+            ('com_github_grpc_grpc', None, ('2.2.0',)),
+            ('com_google_protobuf', 'protobuf-python', ('2.1.0',)),
+            ('com_googlesource_code_re2', None, ('2.1.0',)),
+            ('curl', 'cURL', ('2.1.0',)),
+            ('cython', 'Python', ('2.1.0',)),
+            ('double_conversion', 'double-conversion', ('2.1.0',)),
+            ('enum34_archive', True, ('2.1.0',)),  # Part of Python3
+            ('flatbuffers', 'flatbuffers', ('2.1.0',)),
+            ('functools32_archive', True, ('2.1.0',)),  # Part of Python3
+            ('gast_archive', True, ('2.1.0',)),
+            ('gif', 'giflib', ('2.1.0',)),
+            ('grpc', None, ('2.1.0', '2.2.0')),
+            ('hwloc', 'hwloc', ('2.1.0',)),
+            ('icu', 'ICU', ('2.1.0',)),
+            ('jpeg', 'libjpeg-turbo', ('2.1.0', '2.2.0')),
+            ('jsoncpp_git', 'JsonCpp', ('2.1.0',)),
+            ('keras_applications_archive', True, ('2.1.0', '2.2.0')),
+            ('libjpeg_turbo', 'libjpeg-turbo', ('2.2.0', )),
```

Comment:
@Flamefire Ordering by EasyBuild software name, like it was before, makes more sense to me imho, since we're mapping from EB name (key) to TF name (value).

Maybe it should even be a proper dictionary (with dict values), where you filter the values based on the version range?

```python
available_system_libs = {
    'libjpeg-turbo': {
        ('2.1.0', '2.2.0'): 'jpeg',
        ('2.2.0',): 'libjpeg-turbo',
    },
    ...
    'SWIG': {('2.1.0',): 'swig'},
}
```

or even:

```python
available_system_libs = {
    'libjpeg-turbo': {
        '2.1.0:2.2.0': 'jpeg',
        '2.2.0:': 'libjpeg-turbo',
    },
    ...
    'SWIG': {'2.1.0:': 'swig'},
}
```

Reply:
The dict won't work due to the `True` and `None` values :/ And I'd like to keep them